sup3r/bias/bias_calc.py

```diff
@@ -1155,6 +1174,380 @@ def get_linear_correction(bias_data, base_data, bias_feature, base_dset):
+        Bias correction using QDM.
+    sup3r.preprocessing.data_handling.DataHandler :
+        Bias correction using QDM directly from a derived handler.
+
+    Notes
+    -----
+    One way of using this class is by saving the distributions definitions
+    obtained here with the method `write_outputs()` and then use that file
+    with `local_qdm_bc()` or through a derived `DataHandler`. ATTENTION, be
+    careful handling that file of parameters. There is no checking process
+    and one could apply the correction estimated for the wrong dataset.
+
+    References
+    ----------
+    .. [Cannon2015] Cannon, A. J., Sobie, S. R., & Murdock, T. Q. (2015). Bias
+       correction of GCM precipitation by quantile mapping: how well do
+       methods preserve changes in quantiles and extremes?. Journal of
+       Climate, 28(17), 6938-6959.
+    """
+    def __init__(self,
+                 base_fps,
+                 bias_fps,
+                 bias_fut_fps,
+                 *args,
+                 n_quantiles=101,
+                 dist="empirical",
+                 sampling="linear",
+                 log_base=10,
+                 **kwargs):
+        """Initialize `QuantileDeltaMappingCorrection`
+
+        Parameters
+        ----------
+        base_fps : list | str
+            One or more baseline .h5 filepaths representing non-biased data to
+            use to correct the biased dataset. This is typically several years
+            of WTK or NSRDB files.
+        bias_fps : list | str
+            One or more biased .nc or .h5 filepaths representing the biased
+            data to be compared with the baseline data. This is typically
+            several years of GCM .nc files.
+        bias_fut_fps : list | str
+            Similar to `bias_fps` but usually for a different time period.
+            This is the dataset that would be corrected, since `bias_fsp` is
+            only used to provide a transformation map with the baseline data.
+        dist : str, default="empirical",
+            Define the type of distribution, which can be "empirical" or any
+            parametric distribution defined in "scipy".
+        n_quantiles : int, default=101
+            Defines the number of quantiles for an empirical distribution.
+        sampling : str, default="linear",
+            Defines the spacing bewteen quantiles for an "empirical"
+            distribution.
+        log_base : int or float, default=10,
+            Log base value if sampling is "log" or "invlog".
+
+        See the parent `DataRetrievalBase()` for more possible arguments.
+        """
+
+        self.n_quantiles = n_quantiles
+        self.dist = dist
+        self.sampling = sampling
+        self.log_base = log_base
+
+        super().__init__(base_fps, bias_fps, *args, **kwargs)
+
+        self.bias_fut_fps = bias_fut_fps
+
+        if isinstance(self.bias_fut_fps, str):
+            self.bias_fut_fps = sorted(glob(self.bias_fut_fps))
```

Comment:
@grantbuster , I'm sorry, I didn't understand. Is the issue that the input could be any of:
- "test.h5"
- "*h5"
- ["*h5"]
- ["*h5", "*hdf"]

If that is the issue, would something like below resolve it? If so, should I also update `DataRetrievalBase`?
```
def _expand_paths(fps: str | Sequence):
    if isinstance(fps, str):
        fps = (fps, )

    out = []
    for f in fps:
        out.extend(glob(f))
    return sorted(set(out))
```

Reply:
@castelao I think Grant is suggesting the glob is not needed if the str doesn't include a '*'. But glob will just return [self.bias_fut_fps] if it doesn't include a wildcard anyway, right?

Reply:
@bnb32 , got it! Thanks! I think so. There are also other special characters, such as `?`. Another point of glob is that I believe it only returns items that it can confirm that exist?! I have the feeling that it is safer to let `glob()` just run on whatever.

I added earlier today the `_expand_paths()`, but I don't mind removing it if you both think it would be better.

Reply:
I think it's a useful method. We've implicitly assumed that if we're passing a list its a list of file paths without wildcards but the option to pass a list of globable strings is nice.

Reply:
Nice yeah this works. My original comment was indeed that we should just leave a string alone (don't glob) but i agree passing it through glob is fine because it will make it a list.

Reply:
Great! I'm closing this one.